src/DataMigration/DataMigration.Autorest/custom/Cmdlets/New-AzDataMigrationTdeCertificateMigration.ps1

```diff
@@ -0,0 +1,249 @@
+
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+<#
+.Synopsis
+Migrate TDE certificate from source SQL Server to the target Azure SQL Server.
+.Description
+Migrate TDE certificate from source SQL Server to the target Azure SQL Server.
+#>
+
+
+function New-AzDataMigrationTdeCertificateMigration
+{
+    [OutputType([System.Boolean])]
+    [CmdletBinding(PositionalBinding=$false, ConfirmImpact='Medium')]
+    [Microsoft.Azure.PowerShell.Cmdlets.DataMigration.Description('Migrate TDE certificate from source SQL Server to the target Azure SQL Server.')]
+
+    param(
+        [Parameter(ParameterSetName='CommandLine', Mandatory, HelpMessage='Required. Connection string for the source SQL instance, using the formal connection string format.')]
+        [System.String]
+        ${SourceSqlConnectionString},
+
+        [Parameter(ParameterSetName='CommandLine', Mandatory, HelpMessage='Subscription Id of the target Azure SQL server.')]
+        [System.String]
+        ${TargetSubscriptionId},
+
+        [Parameter(ParameterSetName='CommandLine', Mandatory, HelpMessage='Resource group name of the target Azure SQL server.')]
+        [System.String]
+        ${TargetResourceGroupName},
+
+        [Parameter(ParameterSetName='CommandLine', Mandatory, HelpMessage='Name of the Azure SQL Server.')]
+        [System.String]
+        ${TargetManagedInstanceName},
+
+        [Parameter(ParameterSetName='CommandLine', Mandatory, HelpMessage='Network share path.')]
+        [System.String]
+        ${NetworkSharePath},
+
+        [Parameter(ParameterSetName='CommandLine', Mandatory, HelpMessage='Network share domain.')]
+        [System.String]
+        ${NetworkShareDomain},
+
+        [Parameter(ParameterSetName='CommandLine', HelpMessage='Network share user name.')]
+        [System.String]
+        ${NetworkShareUserName},
+
+        [Parameter(ParameterSetName='CommandLine', HelpMessage='Network share password.')]
+        [System.String]
```

Comment:
For NetworkSharePassword and SourceSqlConnectionString, I initially wanted to use "securestring" to align with other Cmdlets.

However due to what this cmdlet does, it would immediately have to unencrypt the secure string, then pass the values in plaintext to our console app exe.  I believe this defeats the purpose of securestring, but I am wondering if the team has any thoughts on best practices?

Reply:
Recommend to use secure string here anyway